src/target/riscv/field_helpers.h

```diff
@@ -0,0 +1,34 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#ifndef FIELD_HELPERS_H
+#define FIELD_HELPERS_H
+
+#include <stdint.h>
+#include <assert.h>
+
+static inline uint64_t get_field(uint64_t reg, uint64_t mask)
+{
+	return (reg & mask) / (mask & ~(mask << 1));
+}
+
+static inline uint32_t get_field32(uint64_t reg, uint64_t mask)
+{
+	uint64_t value = get_field(reg, mask);
+	assert(value <= UINT32_MAX);
+	return value;
+}
+static inline uint64_t set_field(uint64_t reg, uint64_t mask, uint64_t val)
+{
+	/* Clear current value from field. */
+	reg &= ~mask;
+	uint64_t low_field_bit = mask & ~(mask << 1);
+	reg |= (val * low_field_bit) & mask;
```

Comment:
```suggestion
        assert(!((val * low_field_bit) & ~mask));
	reg |= (val * low_field_bit) & mask;
```

Reply:
Looks like you want to check that the value fits in the mask. Is that right?
Done.

Reply:
> Looks like you want to check that the value fits in the mask. Is that right?
Yes. Thanks!